test/e2e/setup/setup.go

```diff
@@ -46,7 +46,7 @@ func Setup(t *testing.T) *e2eutil.Context {
 		t.Fatal(err)
 	}
 
-	if err := deployOperator(); err != nil {
+	if err := deployOperator(ctx); err != nil {
```

Comment:
I'm passing ctx because before createOrUpdate created an empty context which is why the shouldPerformCleanup flag was not propagated.